### PR TITLE
RDS v3 SDK feature extension

### PR DESCRIPTION
**Pull request by ameszi**
The following RDS v3 API functionalities have been added to the SDK module:
- Instance actions: restart, change storage size, change flavour
- Tag actions: addition and deletion of instance tags

### CODE REVIEW

**Review by gtema — CHANGES_REQUESTED** [review]
Looks already great. There are however few issues:
- you added methods implementation for the resource but not updated the proxy which defines the interface for those (https://github.com/opentelekomcloud/python-otcextensions/blob/master/otcextensions/sdk/rds/v3/_proxy.py). Surely also tests for the proxy need to be added
- we need to have unittest for every function of the resource that we are adding (https://github.com/opentelekomcloud/python-otcextensions/blob/master/otcextensions/tests/unit/sdk/rds/v3/test_instance.py)
- correspondingly documentation for the new functions is missing as well as desired release note (https://github.com/opentelekomcloud/python-otcextensions/blob/master/doc/source/sdk/resources/rds/v3/instance.rst, https://github.com/opentelekomcloud/python-otcextensions/blob/master/doc/source/sdk/proxies/rds_v3.rst and https://github.com/opentelekomcloud/python-otcextensions/tree/master/releasenotes/notes.

I know this is a lot, but we can manage it ;-)

File: otcextensions/sdk/rds/v3/_proxy.py [hunk 1]
```diff
@@ -205,6 +207,69 @@ def set_instance_backup_policy(self, instance, keep_days,
...
+
+        :param instance: This parameter can be either the ID of an instance
+            or a :class:`~openstack.sdk.rds.v3.instance.Instance`
+        :param size:
+        """
+        instance = self._get_resource(_instance.Instance, instance)
+        return instance.enlarge_volume(self, size)
+
+    def change_instance_flavor(self, instance, spec_code):
+        """Chage the instance's flavor.
```
Comment (anton-sidelnikov) [inline otcextensions/sdk/rds/v3/_proxy.py]: Change

File: otcextensions/sdk/rds/v3/_proxy.py [hunk 2]
```diff
@@ -205,6 +207,69 @@ def set_instance_backup_policy(self, instance, keep_days,
...
+        instance = self._get_resource(_instance.Instance, instance)
+        return instance.update_flavor(self, spec_code)
+
+    def get_instance_logs(self, instance, log_type, start_date=None, end_date=None,
+                 offset=1, limit=10, level='ALL'):
+        """Get instance logs. If no dates are specified logs are gathered 
+            from the last 24 hours.
+
+        :param session: The session to use for making this request.
+            :type session: :class:`~keystoneauth1.adapter.Adapter`
```
Comment (anton-sidelnikov) [inline otcextensions/sdk/rds/v3/_proxy.py]: not used

File: otcextensions/sdk/rds/v3/_proxy.py (hunk 1 above)
Comment (ameszi) [reply otcextensions/sdk/rds/v3/_proxy.py]: Fixed.

File: otcextensions/sdk/rds/v3/_proxy.py (hunk 2 above)
Comment (ameszi) [reply otcextensions/sdk/rds/v3/_proxy.py]: Fixed.

File: releasenotes/notes/add-rds-v3-operations-8bb9237548466010.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Add RDS v3 SDK support for the following instance operations: restart, resize volume, change flavor, query logs, add/remove tag.
```
Comment (anton-sidelnikov) [inline releasenotes/notes/add-rds-v3-operations-8bb9237548466010.yaml]: endline missing, all other things are good

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: releasenotes/notes/add-rds-v3-operations-8bb9237548466010.yaml (hunk 3 above)
Comment (ameszi) [reply releasenotes/notes/add-rds-v3-operations-8bb9237548466010.yaml]: Fixed.

**Review by anton-sidelnikov — APPROVED** [review]

File: otcextensions/sdk/rds/v3/_proxy.py [hunk 4]
```diff
@@ -205,6 +207,91 @@ def set_instance_backup_policy(self, instance, keep_days,
...
+        :param int limit: Specifies the number of records on a page.
+            Its value range is from 1 to 100.
+        :param str level: Specifies the log level.
+            Values: ALL, INFO, LOG, WARNING, ERROR, FATAL, PANIC, NOTE.
+        """
+        instance = self._get_resource(_instance.Instance, instance)
+        if log_type not in ['errorlog', 'slowlog']:
+            raise Exception('The parameter log_type has to be either '
+                            '"errorlog" or "slowlog".')
+        if bool(start_date) ^ bool(end_date):
```
Comment (gtema) [inline otcextensions/sdk/rds/v3/_proxy.py:262]: cool

**Review by gtema — APPROVED** [review]

**Pull request merged**